viewer-prototype/src/browser/trace-viewer/components/table-output-component.tsx

```diff
@@ -2,78 +2,177 @@
+    private async fetchTableLines(fetchIndex: number, linesToFetch: number) {
         const traceUUID = this.props.traceId;
         const tspClient = this.props.tspClient;
         const outPutId = this.props.outputDescriptor.id;
 
-        // Fetch columns
-        const columnsResponse = (await tspClient.fetchTableColumns<Entry, EntryHeader>(traceUUID, outPutId, QueryHelper.timeQuery([0, 1]))).getModel();
-        const columnEntries = columnsResponse.model.entries;
-        const columnIds: Array<number> = [];
-        const columnsArray = new Array<any>();
-        columnEntries.forEach(entry => {
-            columnIds.push(entry.id);
-            let columnName = '';
-            if (entry.labels.length) {
-                columnName = entry.labels[0];
-            }
-            columnsArray.push({
-                headerName: columnName,
-                field: entry.id.toString(),
-                width: 200
-            });
-        });
-
-        // Fetch lines
-        const lineResponse = (await tspClient.fetchTableLines(traceUUID, outPutId, QueryHelper.tableQuery(columnIds, 0, 500))).getModel();
+        const lineResponse = (await tspClient.fetchTableLines(traceUUID, outPutId, QueryHelper.tableQuery(this.columnIds, fetchIndex, linesToFetch))).getModel();
         const model = lineResponse.model;
         const lines = model.lines;
         const linesArray = new Array<any>();
         lines.forEach(line => {
             const obj: any = {};
             const cells = line.cells;
             const ids = model.columnIds;
+
+            if (this.showIndexColumn) {
+                obj[0] = line.index.toString();
+            }
+
             for (let i = 0; i < cells.length; i++) {
-                obj[ids[i]] = cells[i].content;
+                const id = this.showIndexColumn ? ids[i] + 1 : ids[i];
+                obj[id] = cells[i].content;
             }
             linesArray.push(obj);
         });
 
         this.setState({
-            outputStatus: lineResponse.status,
-            tableColumns: columnsArray,
-            tableLines: linesArray
+            tableColumns: this.columnArray,
+            outputStatus: status
         });
+
+        return linesArray;
+    }
+
+    private onGridReady = async (event: GridReadyEvent) => {
+        const dataSource: IDatasource = {
+            getRows: async params => {
+                await this.init();
+                const rowsThisPage = await this.fetchTableLines(params.startRow, params.endRow - params.startRow);
+                for (let i = 0; i < rowsThisPage.length; i++) {
+                    const item = rowsThisPage[i];
+                    const itemCopy = cloneDeep(item);
+                    rowsThisPage[i] = itemCopy;
+                }
+
+                if (rowsThisPage.length < this.cacheBlockSize) {
+                    params.successCallback(rowsThisPage, params.startRow + rowsThisPage.length);
+                }
+                else {
```

Comment:
else should be in same line as closing bracket

Reply:
Done.